appserver/web/weld-integration/src/main/java/org/glassfish/cdi/transaction/TransactionalInterceptorRequiresNew.java

```diff
@@ -132,16 +132,16 @@ public Object transactional(InvocationContext ctx) throws Exception {
                             "Managed bean with Transactional annotation and TxType of REQUIRES_NEW " +
                             "encountered exception during commit " + exception,
                             exception);
-                }
-
-                if (suspendedTransaction != null) {
-                    try {
-                        getTransactionManager().resume(suspendedTransaction);
-                    } catch (Exception exception) {
-                        throw new TransactionalException(
-                                "Managed bean with Transactional annotation and TxType of REQUIRED " +
-                                "encountered exception during resume " + exception,
-                                exception);
+                } finally {
```

Comment:
SonarQube has a rule for this - this would mask the possible exception from the try block, because finally is executed always. See https://www.oracle.com/technical-resources/articles/java/trywithresources.html

Reply:
The exception from proceed would also be masked, so I changed the flow to keep the original exceptions as suppressed, throwing the last exception.